src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsController.java

```diff
@@ -320,4 +327,26 @@ public ResponseEntity<ErrorMessage> handleNoRoleAssignmentsException(Exception e
             ));
     }
 
+    private Optional<AccessControlResponse> getAssigneeAccessControlResponse(
+        String assignerAuthToken,
+        AssignTaskRequest assignTaskRequest,
+        AccessControlResponse assignerAccessControlResponse) {
+        UserInfo userInfo = assignerAccessControlResponse.getUserInfo();
+        return assignTaskRequest.getUserId() == null
+            && isGranularPermissionFeatureEnabled(userInfo.getUid(), userInfo.getEmail())
```

Comment:
we should at least have an IT scenario for this.

Reply:
actioned